lib/pages/preferences_page.dart

```diff
@@ -10,102 +11,80 @@ class PreferencesPage extends StatefulWidget {
 }
 
 class _PreferencesPageState extends State<PreferencesPage> {
-  Future<SharedPreferences> _preferences;
 
   @override
   void initState() {
     super.initState();
-    _preferences = SharedPreferences.getInstance();
   }
 
   @override
   Widget build(BuildContext context) {
-    return ThemeSwitchingArea(
-      child: Scaffold(
-        backgroundColor: Theme.of(context).primaryColor,
-        appBar: AppBar(title: Text("Préférences")),
-        body: _buildPreferencesContent(),
-      ),
+    return Scaffold(
+      backgroundColor: Theme.of(context).primaryColor,
+      appBar: AppBar(title: Text("Préférences")),
+      body: _buildPreferencesContent(),
     );
   }
 
   Widget _buildPreferencesContent() {
-    return FutureBuilder<SharedPreferences>(
-      future: _preferences,
-      builder: (context, snapshot) {
-        switch (snapshot.connectionState) {
-          case ConnectionState.waiting:
-            return Center(
-              child: Column(
+    return BlocBuilder<PreferencesBloc, PreferencesState>(
+      bloc: BlocProvider.of<PreferencesBloc>(context),
```

Comment:
I'm not sure you have to specify the bloc here! The `BlocBuilder` should be able to retrieve the `PreferenceBloc` instance from the inherited context because you specified its type in its declaration (i.e. `BlocBuilder<PreferenceBloc, xxx>`). No ?